src/abstractdataframe/prettytables.jl

```diff
@@ -0,0 +1,142 @@
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+#
+# Description
+# ==============================================================================
+#
+#     This file contains the functions that build the interface with
+#     PrettyTables.jl.
+#
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+using PrettyTables
+
+# Internal function to print DataFrames using PrettyTables.jl.
+function _pretty_table(io::IO, df::AbstractDataFrame;
+                       allrows::Bool = !get(io, :limit, false),
+                       allcols::Bool = !get(io, :limit, false),
+                       rowlabel::Symbol = :Row,
+                       summary::Bool = true,
+                       eltypes::Bool = true,
+                       rowid=nothing,
+                       truncstring::Int = 32,
+                       kwargs...)
+
+    names = reshape( propertynames(df), (1,:) )
+    types = DataFrames.compacttype.(reshape( eltype.(eachcol(df)), (1,:) ))
+
+    crop = :both
+
+    if allcols && allrows
+        crop = :none
+    elseif allcols
+        crop = :vertical
+    elseif allrows
+        crop = :horizontal
+    end
+
+    # Update the maximum column width. This is necessary because the default
+    # formatter must have access to the option we are using.
+    _PRETTY_TABLES_CONF[:maximum_columns_width]     = truncstring
+    _PRETTY_TABLES_SAFECONF[:maximum_columns_width] = truncstring
+
+    # Assemble the configurations for this print.
+    _local_conf = deepcopy(_PRETTY_TABLES_CONF)
+
+    for kw in kwargs
+        _local_conf[kw[1]] = kw[2]
+    end
+
+    # Check if the user wants to display a summary about the DataFrame that is
+    # being printed. This will be shown using the `title` option of
+    # `pretty_table`.
+    title = summary ? Base.summary(df) : ""
+
+    # Transform into a named tuple so that it can be passed to PrettyTables.jl.
+    dictkeys = (collect(keys(_local_conf))...,)
+    dictvals = (collect(values(_local_conf))...,)
+    nt = NamedTuple{dictkeys}(dictvals)
+
+    # Print the table with the selected options.
+    try
+        pretty_table(io, df,
+                     vcat(names,types);
+                     crop = crop,
+                     nosubheader = !eltypes,
+                     row_number_column_title = string(rowlabel),
+                     title = title,
+                     nt...)
+    catch
+        @warn """An unsupported argument was passed to PrettyTables.jl.
```

Comment:
Better make this an error. This simplifies the code and there's no reason to accept invalid arguments.

Reply:
Done!